src/Components/SingleArticle.js

```diff
@@ -0,0 +1,35 @@
+import { useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { fetchArticleById } from "../api";
+
+export default function SingleArticle() {
+  const [article, setArticle] = useState({});
+  const { article_id } = useParams();
+
+  useEffect(() => {
+    fetchArticleById(article_id).then(({ article }) => {
+      setArticle(article);
```

Comment:
Rather than running setArticle(article) and then referring to article[0] throughout the component, why not do setArticle(article[0])? Then your state doesn't change data type and your code is more reliable :)